src/cascadia/TerminalApp/App.cpp

```diff
@@ -3,667 +3,60 @@
-        dialog.XamlRoot(_root->XamlRoot());
-
-        // IMPORTANT: Set the requested theme of the dialog, because the
-        // PopupRoot isn't directly in the Xaml tree of our root. So the dialog
-        // won't inherit our RequestedTheme automagically.
-        dialog.RequestedTheme(_settings->GlobalSettings().GetRequestedTheme());
-
-        // Display the dialog.
-        co_await dialog.ShowAsync(Controls::ContentDialogPlacement::Popup);
-
-        // After the dialog is dismissed, the dialog lock (held by `lock`) will
-        // be released so another can be shown
+        Initialize();
     }
 
-    // Method Description:
-    // - Displays a dialog for errors found while loading or validating the
-    //   settings. Uses the resources under the provided  title and content keys
-    //   as the title and first content of the dialog, then also displays a
-    //   message for whatever exception was found while validating the settings.
-    // - Only one dialog can be visible at a time. If another dialog is visible
-    //   when this is called, nothing happens. See _ShowDialog for details
-    // Arguments:
-    // - titleKey: The key to use to lookup the title text from our resources.
-    // - contentKey: The key to use to lookup the content text from our resources.
-    void App::_ShowLoadErrorsDialog(const winrt::hstring& titleKey,
-                                    const winrt::hstring& contentKey,
-                                    HRESULT settingsLoadedResult)
+    AppLogic App::Logic()
     {
-        auto title = GetLibraryResourceString(titleKey);
-        auto buttonText = RS_(L"Ok");
-
-        Controls::TextBlock warningsTextBlock;
-        // Make sure you can copy-paste
-        warningsTextBlock.IsTextSelectionEnabled(true);
-        // Make sure the lines of text wrap
-        warningsTextBlock.TextWrapping(TextWrapping::Wrap);
-
-        winrt::Windows::UI::Xaml::Documents::Run errorRun;
-        const auto errorLabel = GetLibraryResourceString(contentKey);
-        errorRun.Text(errorLabel);
-        warningsTextBlock.Inlines().Append(errorRun);
-
-        if (FAILED(settingsLoadedResult))
-        {
-            if (!_settingsLoadExceptionText.empty())
-            {
-                warningsTextBlock.Inlines().Append(_BuildErrorRun(_settingsLoadExceptionText, Resources()));
-            }
-        }
-
-        // Add a note that we're using the default settings in this case.
-        winrt::Windows::UI::Xaml::Documents::Run usingDefaultsRun;
-        const auto usingDefaultsText = RS_(L"UsingDefaultSettingsText");
-        usingDefaultsRun.Text(usingDefaultsText);
-        warningsTextBlock.Inlines().Append(usingDefaultsRun);
-
-        Controls::ContentDialog dialog;
-        dialog.Title(winrt::box_value(title));
-        dialog.Content(winrt::box_value(warningsTextBlock));
-        dialog.CloseButtonText(buttonText);
-
-        _ShowDialog(nullptr, dialog);
+        static AppLogic logic;

```

Comment:
Why is this not a member of `App`? You could wait to instantiate it until this is first called.

Reply:
@duhowett-msft, as far as I can see, the magic static is safer than the alternative as it cannot accidentally be referenced from elsewhere in the class without being allocated and will only be allocated deferred to the first call.

Doing what Mike says opens us up to a null check on the instance variable, theoretically. But this is indeed magic and maybe we shouldn't magic?